### PR TITLE
fix(forge): BaseCounterExample for targetInterface

### PR DESCRIPTION
**Pull request by bernard-wagner**
## Motivation

Resolves https://github.com/foundry-rs/foundry/issues/5891

## Solution

Do not bomb on failing to create a counter example, rather just return raw trace. This seems like the a more suitable behaviour, regardless of the issues introduced with `targetInterface`. 

This is still not perfect, as we would want to decode the input when using `targetInterface`, but that would require a lot more changes to propagate the associated interfaces to the tracer.

### CODE REVIEW

File: crates/evm/src/fuzz/mod.rs [hunk 1]
```diff
@@ -301,25 +301,33 @@ impl BaseCounterExample {
...
-
-        let func = abi
-            .functions()
-            .find(|f| f.short_signature() == bytes.0.as_ref()[0..4])
-            .ok_or(FuzzError::UnknownFunction)?;
-
-        // skip the function selector when decoding
-        let args =
-            func.decode_input(&bytes.0.as_ref()[4..]).map_err(|_| FuzzError::FailedDecodeInput)?;
+        if let Some((name, abi)) = &contracts.get(&addr) {
```
Comment (mattsse) [inline crates/evm/src/fuzz/mod.rs:304]: this now always returns Ok?

is this intended?

**Review by mattsse — COMMENTED** [review]
really not familiar with the invariant internals, so can't really comment on it since I don't fully understand it yet

File: crates/evm/src/fuzz/mod.rs (hunk 1 above)
Comment (bernard-wagner) [reply crates/evm/src/fuzz/mod.rs:304]: Yes, was always returning Ok. Removed unnecessary Result. Just returns Self now.

**Review by mattsse — APPROVED** [review]
I think I got it now, should resolve at least the mentioned issue

**Pull request merged**